controllers/storagecluster/storageclasses.go

```diff
@@ -231,6 +245,38 @@ func newCephBlockPoolStorageClassConfiguration(initData *ocsv1.StorageCluster) S
 	}
 }
 
+// newCephNFSStorageClassConfiguration generates configuration options for a Ceph Filesystem StorageClass.
+func newCephNFSStorageClassConfiguration(initData *ocsv1.StorageCluster) StorageClassConfiguration {
+	persistentVolumeReclaimDelete := corev1.PersistentVolumeReclaimDelete
+	// VolumeExpansion is not yet supported for nfs.
+	allowVolumeExpansion := false
+	return StorageClassConfiguration{
+		storageClass: &storagev1.StorageClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: generateNameForCephNetworkFilesystemSC(initData),
+				Annotations: map[string]string{
+					"description": "Provides RWO and RWX Filesystem volumes",
+				},
+			},
+			Provisioner:          generateNameForNFSCSIProvisioner(initData),
+			ReclaimPolicy:        &persistentVolumeReclaimDelete,
+			AllowVolumeExpansion: &allowVolumeExpansion,
+			Parameters: map[string]string{
+				"clusterID":  initData.Namespace,
+				"nfsCluster": generateNameForCephNFS(initData),
+				"fsName":     generateNameForCephFilesystem(initData),
+				"server":     generateNameForNFSService(initData),
```

Comment:
It would be good to set a volume prefix, otherwise it will inherit the default from the CephFS provisioner. In Ceph-CSI the example sets it to `volumeNamePrefix: nfs-export-`.

Reply:
> It would be good to set a volume prefix, otherwise it will inherit the default from the CephFS provisioner. In Ceph-CSI the example sets it to `volumeNamePrefix: nfs-export-`.

Makes sense to me.
@Madhu-1 @BlaineEXE wdyt?

Reply:
sounds good to me, make sure this is well tested. `volumeNamePrefix` is not very well tested.

Reply:
> sounds good to me, make sure this is well tested. `volumeNamePrefix` is not very well tested.

There is e2e for this at cephcsi ci https://github.com/ceph/ceph-csi/blob/2b71aac752614182fb0ba5dd956307401d2920b1/e2e/cephfs.go#L420 
Anything else needs to be tested?

Reply:
added this now

Reply:
Yes, we added this in upstream long back. It's only CI tested just to make sure some corner cases like provisioner restart etc are well tested.